File: excellence/packages/extraction-core/src/models/ValidationResult.ts
export type IssueSeverity = 'error' | 'warning' | 'info';
export type IssueType =
	| 'type_mismatch'
	| 'sum_discrepancy'
	| 'duplicate_detected'
	| 'missing_value'
	| 'ocr_low_confidence'
	| 'overwrite_risk'
	| 'cross_field_violation';

export interface ValidationIssue {
	type: IssueType;
	row?: number;
	col?: number;
	message: string;
	severity: IssueSeverity;
}

export interface SumReconciliationResult {
	columnName: string;
	sourceTotal: number;
	computedTotal: number;
	match: boolean;
	discrepancy: number;
}

export interface DuplicateDetectionResult {
	duplicateFound: boolean;
	existingSheetName?: string;
	similarityScore?: number;
}

export interface ValidationResult {
	jobId: string;
	passed: boolean;
	issues: ValidationIssue[];
	sumReconciliation?: SumReconciliationResult;
	duplicateDetection?: DuplicateDetectionResult;
}

File: excellence/services/api/models/validation_result.py
from typing import List, Literal, Optional

from .base import CamelModel


IssueSeverity = Literal["error", "warning", "info"]
IssueType = Literal[
    "type_mismatch",
    "sum_discrepancy",
    "duplicate_detected",
    "missing_value",
    "ocr_low_confidence",
    "overwrite_risk",
    "cross_field_violation",
]


class ValidationIssue(CamelModel):
    type: IssueType
    row: Optional[int] = None
    col: Optional[int] = None
    message: str
    severity: IssueSeverity


class SumReconciliationResult(CamelModel):
    column_name: str
    source_total: float
    computed_total: float
    match: bool
    discrepancy: float


class DuplicateDetectionResult(CamelModel):
    duplicate_found: bool
    existing_sheet_name: Optional[str] = None
    similarity_score: Optional[float] = None


class ValidationResult(CamelModel):
    job_id: str
    passed: bool
    issues: List[ValidationIssue]
    sum_reconciliation: Optional[SumReconciliationResult] = None
    duplicate_detection: Optional[DuplicateDetectionResult] = None
